### DISCUSSION TITLE
Preserve Documentation Website Sidebar's Scroll State

### DISCUSSION THREAD

**Discussion by brannytempy** [twilio-labs/paste#2732] [Ideas] [answered]
I've been using the Paste site a lot and LOVE it!! One gripe I have is how scroll isn't preserved between pages. As I navigate to each page, the scroll position is reset. See example video below:

https://user-images.githubusercontent.com/1590723/195638241-914ca80e-21c3-4400-b3f6-9ce669c5ed66.mp4

This ends up creating a lot of disorientation when I am perusing the documentation from top to bottom. The Paste site is about to get a lot of traffic from Segment as we migrate, and this improvement might be helpful to others who are exploring without aim.

**Comment by TheSisb** [comment, marked as answer]
Hey @brannytempy,

Thanks for the feedback! This has been [recommended once before too](https://github.com/twilio-labs/paste/discussions/2346), and we're planning on making this improvement over the next few months as we migrate our doc site off Gatsby and onto NextJS.

**Reply by nkrantz** [reply]
Fell free to follow along in JIRA: https://issues.corp.twilio.com/browse/DSYS-3308
